### PR TITLE
Refactor： use [fusio](https://github.com/tonbo-io/fusio) to implement the storage layer and support multiple storage

### PR DESCRIPTION
**Pull request by KKould**

### CODE REVIEW

File: src/serdes/num.rs [hunk 1]
```diff
@@ -23,16 +23,13 @@ macro_rules! implement_encode_decode {
...
-                let buf = {
-                    let mut buf = [0; size_of::<Self>()];
-                    reader.read_exact(&mut buf).await?;
-                    buf
-                };
+            async fn decode<R: Read + Unpin>(reader: &mut R) -> Result<Self, Self::Error> {
+                let bytes = reader.read_exact(vec![0u8; size_of::<Self>()]).await?;
 
-                Ok(Self::from_le_bytes(buf))
+                // SAFETY
```
Comment (ethe) [inline src/serdes/num.rs:31]: useless safety condition?

**Review by ethe — APPROVED** [review]

File: src/serdes/num.rs [hunk 2]
```diff
@@ -23,16 +23,13 @@ macro_rules! implement_encode_decode {
...
+            type Error = fusio::Error;
 
-            async fn decode<R: AsyncRead + Unpin>(reader: &mut R) -> Result<Self, Self::Error> {
-                let buf = {
-                    let mut buf = [0; size_of::<Self>()];
-                    reader.read_exact(&mut buf).await?;
-                    buf
-                };
+            async fn decode<R: Read + Unpin>(reader: &mut R) -> Result<Self, Self::Error> {
+                let bytes = reader.read_exact(vec![0u8; size_of::<Self>()]).await?;
```
Comment (ethe) [inline src/serdes/num.rs:29]: use array instead of vector?

File: src/version/mod.rs [hunk 3]
```diff
@@ -113,42 +107,56 @@ where
...
-                .table_query(key, &scope.gen, projection_mask.clone())
+                .table_query(level_0_fs, key, &scope.gen, projection_mask.clone())
                 .await?
             {
                 return Ok(Some(entry));
             }
         }
-        for level in self.level_slice[1..6].iter() {
-            if level.is_empty() {
+        for (i, sort_runs) in self.level_slice[1..6].iter().enumerate() {
```
Comment (crwen) [inline src/version/mod.rs:140]: should it be `level_slice[1..MAX_LEVEL]`? level_slice[6] is not included in this case

File: src/executor.rs [hunk 4]
```diff
@@ -1,8 +1,6 @@
 use std::future::Future;
 
-use crate::fs::FileProvider;
-
-pub trait Executor: FileProvider {
+pub trait Executor {
```
Comment (ethe) [inline src/executor.rs:3]: Executor has to provide the local file system implementation, use tokio executor + monoio filesystem should cause panic.

**Review by crwen — APPROVED** [review]

**Pull request merged**